### PR TITLE
Use the fake time instead of real in tests

### LINKED ISSUE #2287: [kueuectl] Use the fake time instead of real in tests
**What would you like to be cleaned**:
Introduce the same approach as controllers like this:

https://github.com/kubernetes-sigs/kueue/blob/4b9744b29bec33bbd34e01a66f09002b478a6493/pkg/controller/core/workload_controller.go#L117

https://github.com/kubernetes-sigs/kueue/blob/4b9744b29bec33bbd34e01a66f09002b478a6493/pkg/controller/core/workload_controller_test.go#L807

Also check [this](https://github.com/kubernetes-sigs/kueue/pull/2195#discussion_r1607067637).

**Why is this needed**:

### LINKED ISSUE #2353: Flaky integration test: Should print workloads list with paging
/kind flake

**What happened**:

Integration test failed: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/2314/pull-kueue-test-integration-main/1798039851593895936

**What you expected to happen**:

No flake

**How to reproduce it (as minimally and precisely as possible)**:

Repeat the build.

**Anything else we need to know?**:

```
  [FAILED] Expected
      <string>: "...     1s
      "
  to equal               |
      <string>: "...     0s
      "
  In [It] at: /home/prow/go/src/sigs.k8s.io/kueue/test/integration/kueuectl/list_test.go:206 @ 06/04/24 17:19:10.001
```

### PR DESCRIPTION
**Pull request by mszadkow**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
We want to use fake clocks in tests as they allow for necessary time manipulation when required.

#### Which issue(s) this PR fixes:

Fixes #2287
Fixes #2353

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cmd/kueuectl/app/cmd.go [hunk 1]
```diff
@@ -37,6 +38,7 @@ type KueuectlOptions struct {
 	ConfigFlags *genericclioptions.ConfigFlags
 
 	genericiooptions.IOStreams
+	Clock clock.Clock
```
Comment (mbobrovskyi) [inline cmd/kueuectl/app/cmd.go:41]:
```suggestion
        Clock clock.Clock
	ConfigFlags *genericclioptions.ConfigFlags

	genericiooptions.IOStreams
```

File: cmd/kueuectl/app/list/list_clusterqueue.go [hunk 2]
```diff
@@ -48,6 +49,7 @@ type ClusterQueueOptions struct {
 	Limit         int64
 	LabelSelector string
 	FieldSelector string
+	Clock         clock.Clock
```
Comment (mbobrovskyi) [inline cmd/kueuectl/app/list/list_clusterqueue.go:52]: Maybe move it to 46 line? And on other lists too?

File: cmd/kueuectl/app/list/list_workload_test.go [hunk 3]
```diff
@@ -567,7 +568,8 @@ wl2               j2         lq2          cq2            PENDING   22
 			tf.KueueClientset = clientset
 			tf.KueueClientset.Discovery().(*fakediscovery.FakeDiscovery).Resources = tc.apiResourceLists
 
-			cmd := NewWorkloadCmd(tf, streams)
+			fC := testingclock.NewFakeClock(testStartTime)
```
Comment (mbobrovskyi) [inline cmd/kueuectl/app/list/list_workload_test.go:571]:
```suggestion
			fc := testingclock.NewFakeClock(testStartTime)
```

**Review by mbobrovskyi — COMMENTED** [review]
Overall LGTM. Only few nits.

File: cmd/kueuectl/app/list/list_clusterqueue.go [hunk 4]
```diff
@@ -117,7 +120,8 @@ func (o *ClusterQueueOptions) Complete(clientGetter util.ClientGetter, cmd *cobr
 func (o *ClusterQueueOptions) ToPrinter(headers bool) (printers.ResourcePrinterFunc, error) {
 	if !o.PrintFlags.OutputFlagSpecified() {
 		printer := newClusterQueueTablePrinter().
-			WithHeaders(headers)
+			WithHeaders(headers).
+			WithClock(o.Clock)
```
Comment (mbobrovskyi) [inline cmd/kueuectl/app/list/list_clusterqueue.go:124]: Do you have default implementation of clock inside printer? What happens if we will not set `WithClock(o.Clock)`?

**Comment by mimowo** [comment]
Please also update the PR description to note that it fixes https://github.com/kubernetes-sigs/kueue/issues/2353

File: cmd/kueuectl/app/list/list_clusterqueue.go (hunk 4 above)
Comment (mszadkow) [reply cmd/kueuectl/app/list/list_clusterqueue.go:124]: You are right, I am currently working on this.
I think, that's the reason some tests fail.

File: test/integration/kueuectl/list_test.go [hunk 5]
```diff
@@ -70,7 +71,7 @@ var _ = ginkgo.Describe("Kueuectl List", ginkgo.Ordered, ginkgo.ContinueOnFailur
 		ginkgo.It("Should print local queues list filtered by field selector", func() {
 			streams, _, output, errOutput := genericiooptions.NewTestIOStreams()
 			configFlags := CreateConfigFlagsWithRestConfig(cfg, streams)
-			kueuectl := app.NewKueuectlCmd(app.KueuectlOptions{ConfigFlags: configFlags, IOStreams: streams})
+			kueuectl := app.NewKueuectlCmd(app.KueuectlOptions{ConfigFlags: configFlags, IOStreams: streams, Clock: testingclock.NewFakeClock(time.Now())})
```
Comment (mbobrovskyi) [inline test/integration/kueuectl/list_test.go:74]: IIUC we need to set `executeTime := time.Now()` from L78 here.

File: cmd/kueuectl/app/list/list_clusterqueue_printer.go [hunk 6]
```diff
@@ -19,23 +19,28 @@ package list
...
 
 type listClusterQueuePrinter struct {
+	clock        clock.Clock
 	printOptions printers.PrintOptions
 }
 
 var _ printers.ResourcePrinter = (*listClusterQueuePrinter)(nil)
 
 func (p *listClusterQueuePrinter) PrintObj(obj runtime.Object, out io.Writer) error {
+	if p.clock == nil {
```
Comment (mbobrovskyi) [inline cmd/kueuectl/app/list/list_clusterqueue_printer.go:40]: I think it should enough to have it on `newClusterQueueTablePrinter`.

**Pull request merged**